### ISSUE TITLE
[AC] [Bug] Cannot join WSG in a NPCBOT Party

### ISSUE THREAD

**Issue by Badgermilk0** [trickerer/Trinity-Bots#918]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that my issue may get **closed without notice** if I *intentionally* remove or skip any mandatory\* field

### Current behaviour

For Battlegrounds:
I CANNOT join WSG while in a NPCBot party.

I CAN join the other 3 supported ones while in a NPCBot party.

### TrinityCore or AzerothCore

AzerothCore

### Core rev. hash/commit

AzerothCore rev. 64cea460e809 2024-11-07 17:32:05 +0700 (npcbots_3.3.5 branch)

### Operating system

Win 11

**Comment by Badgermilk0** [comment]
A potential cause.
I **_think_** the character was in a dungeon with NPCbots when the "NpcBot.OwnershipExpireTime" ran out (while offline).

**Comment by trickerer** [comment]
After doing a dungeon using LFG you have to re-make your party, otherwise it will be considered a LFG group and will not allow joining BG queue, is that the problem?

**Comment by Badgermilk0** [comment]
When I use the "Join as Group" for Warsong Gulch when in a NPCBot party, I get placed in a never-ending queue:

![image](https://github.com/user-attachments/assets/1a617439-164f-4db2-9ce3-6bc56a16c2e9)

It works for all the other supported Battle Grounds.
ex. Alterac Valley pops right away while in a NPCbot Party

![image](https://github.com/user-attachments/assets/d3da196f-8cfb-44d9-8ecf-25050dd0dfe3)

**Comment by trickerer** [comment]
This can happen in two cases:
a) there is some other group already in that BG with generated bots (only one can be active at a time per BG/bracket combo)
b) you queue into WSG as a 5-man group and you `NpcBot.Premade.Enable` is set to `1` in config - in that case bots won't be generated as premade groups are only suppposed to battle with each other

Or maybe this is a bug and queue has hanged or something. Does server restart solve it?

If none of the above, open the config and set npcbots log level to debug (`Logger.npcbots=2,NpcBots Server`), restart the server and post the console output after trying to queue for WSG

**Comment by Badgermilk0** [comment]
I have "NpcBot.Premade.Enable = 1" and "Logger.npcbots=5,NpcBots Server"
I am in a premade NPCBot Party of 4+me and then click "Join as Group" for Warsong Gulch.
The "You are in a queue for Warsong Gulch" pops up but no bots join, nothing gets output to the Worldserver CLI.

If I click "Join as Group" for Arathi Basin, a BattleGround is created immediatly and I can join with me NPCbot party.

For WSG Log:
Nothing gets output to server.log

For Arathi Basin Log:
Spawning wandering bot: Sheldras (81049) class 11 race 4 fac 4, location: mapId 529 WP 5025 'ArathiBasinSpawnA', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3417, minLvl 20, maxLvl 80 (X: 1289.51 Y: 1286.41 Z: -14.47), flags: 0x00000013 (ArathiBasinSpawnA)
BG bot Sheldras id 81049 selected level 57...
Spawning wandering bot: Killac (81050) class 3 race 11 fac 1629, location: mapId 529 WP 5025 'ArathiBasinSpawnA', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3417, minLvl 20, maxLvl 80 (X: 1289.51 Y: 1286.41 Z: -14.47), flags: 0x00000013 (ArathiBasinSpawnA)
BG bot Killac id 81050 selected level 52...
Spawning wandering bot: Lyria (81051) class 1 race 1 fac 1, location: mapId 529 WP 5025 'ArathiBasinSpawnA', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3417, minLvl 20, maxLvl 80 (X: 1289.51 Y: 1286.41 Z: -14.47), flags: 0x00000013 (ArathiBasinSpawnA)
BG bot Lyria id 81051 selected level 51...
Spawning wandering bot: Caedmos (81052) class 5 race 11 fac 1629, location: mapId 529 WP 5025 'ArathiBasinSpawnA', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3417, minLvl 20, maxLvl 80 (X: 1289.51 Y: 1286.41 Z: -14.47), flags: 0x00000013 (ArathiBasinSpawnA)
BG bot Caedmos id 81052 selected level 52...
Spawning wandering bot: Brandur (81053) class 2 race 3 fac 3, location: mapId 529 WP 5025 'ArathiBasinSpawnA', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3417, minLvl 20, maxLvl 80 (X: 1289.51 Y: 1286.41 Z: -14.47), flags: 0x00000013 (ArathiBasinSpawnA)
BG bot Brandur id 81053 selected level 59...
Spawning wandering bot: Thorfin (81054) class 3 race 3 fac 3, location: mapId 529 WP 5025 'ArathiBasinSpawnA', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3417, minLvl 20, maxLvl 80 (X: 1289.51 Y: 1286.41 Z: -14.47), flags: 0x00000013 (ArathiBasinSpawnA)
BG bot Thorfin id 81054 selected level 56...
Spawning wandering bot: Benjamin (81055) class 5 race 1 fac 1, location: mapId 529 WP 5025 'ArathiBasinSpawnA', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3417, minLvl 20, maxLvl 80 (X: 1289.51 Y: 1286.41 Z: -14.47), flags: 0x00000013 (ArathiBasinSpawnA)
BG bot Benjamin id 81055 selected level 55...
Spawning wandering bot: Keilnei (81056) class 3 race 11 fac 1629, location: mapId 529 WP 5025 'ArathiBasinSpawnA', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3417, minLvl 20, maxLvl 80 (X: 1289.51 Y: 1286.41 Z: -14.47), flags: 0x00000013 (ArathiBasinSpawnA)
BG bot Keilnei id 81056 selected level 57...
Spawning wandering bot: Kore (81057) class 1 race 11 fac 1629, location: mapId 529 WP 5025 'ArathiBasinSpawnA', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3417, minLvl 20, maxLvl 80 (X: 1289.51 Y: 1286.41 Z: -14.47), flags: 0x00000013 (ArathiBasinSpawnA)
BG bot Kore id 81057 selected level 56...
Spawning wandering bot: Stefan (81058) class 6 race 1 fac 1, location: mapId 529 WP 5025 'ArathiBasinSpawnA', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3417, minLvl 20, maxLvl 80 (X: 1289.51 Y: 1286.41 Z: -14.47), flags: 0x00000013 (ArathiBasinSpawnA)
BG bot Stefan id 81058 selected level 57...
Spawning wandering bot: Jorik (81059) class 4 race 1 fac 1, location: mapId 529 WP 5025 'ArathiBasinSpawnA', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3417, minLvl 20, maxLvl 80 (X: 1289.51 Y: 1286.41 Z: -14.47), flags: 0x00000013 (ArathiBasinSpawnA)
BG bot Jorik id 81059 selected level 56...
Spawning wandering bot: Azar (81060) class 2 race 3 fac 3, location: mapId 529 WP 5025 'ArathiBasinSpawnA', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3417, minLvl 20, maxLvl 80 (X: 1289.51 Y: 1286.41 Z: -14.47), flags: 0x00000013 (ArathiBasinSpawnA)
BG bot Azar id 81060 selected level 51...
Spawning wandering bot: Nara (81061) class 5 race 4 fac 4, location: mapId 529 WP 5025 'ArathiBasinSpawnA', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3417, minLvl 20, maxLvl 80 (X: 1289.51 Y: 1286.41 Z: -14.47), flags: 0x00000013 (ArathiBasinSpawnA)
BG bot Nara id 81061 selected level 54...
Spawning wandering bot: Sandahl (81062) class 9 race 1 fac 1, location: mapId 529 WP 5025 'ArathiBasinSpawnA', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3417, minLvl 20, maxLvl 80 (X: 1289.51 Y: 1286.41 Z: -14.47), flags: 0x00000013 (ArathiBasinSpawnA)
BG bot Sandahl id 81062 selected level 56...
Spawning wandering bot: Ormyr (81063) class 4 race 3 fac 3, location: mapId 529 WP 5025 'ArathiBasinSpawnA', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3417, minLvl 20, maxLvl 80 (X: 1289.51 Y: 1286.41 Z: -14.47), flags: 0x00000013 (ArathiBasinSpawnA)
BG bot Ormyr id 81063 selected level 56...
Spawning wandering bot: Grezz (81064) class 1 race 2 fac 2, location: mapId 529 WP 5035 'ArathiBasinSpawnH', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3418, minLvl 20, maxLvl 80 (X: 704.07 Y: 704.46 Z: -16.43), flags: 0x00000015 (ArathiBasinSpawnH)
BG bot Grezz id 81064 selected level 55...
Spawning wandering bot: Duesten (81065) class 5 race 5 fac 5, location: mapId 529 WP 5035 'ArathiBasinSpawnH', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3418, minLvl 20, maxLvl 80 (X: 704.07 Y: 704.46 Z: -16.43), flags: 0x00000015 (ArathiBasinSpawnH)
BG bot Duesten id 81065 selected level 56...
Spawning wandering bot: Gest (81066) class 4 race 2 fac 2, location: mapId 529 WP 5035 'ArathiBasinSpawnH', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3418, minLvl 20, maxLvl 80 (X: 704.07 Y: 704.46 Z: -16.43), flags: 0x00000015 (ArathiBasinSpawnH)
BG bot Gest id 81066 selected level 58...
Spawning wandering bot: Oninath (81067) class 3 race 10 fac 1610, location: mapId 529 WP 5035 'ArathiBasinSpawnH', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3418, minLvl 20, maxLvl 80 (X: 704.07 Y: 704.46 Z: -16.43), flags: 0x00000015 (ArathiBasinSpawnH)
BG bot Oninath id 81067 selected level 51...
Spawning wandering bot: Anastasia (81068) class 8 race 5 fac 5, location: mapId 529 WP 5035 'ArathiBasinSpawnH', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3418, minLvl 20, maxLvl 80 (X: 704.07 Y: 704.46 Z: -16.43), flags: 0x00000015 (ArathiBasinSpawnH)
BG bot Anastasia id 81068 selected level 50...
Spawning wandering bot: Kym (81069) class 11 race 6 fac 6, location: mapId 529 WP 5035 'ArathiBasinSpawnH', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3418, minLvl 20, maxLvl 80 (X: 704.07 Y: 704.46 Z: -16.43), flags: 0x00000015 (ArathiBasinSpawnH)
BG bot Kym id 81069 selected level 60...
Spawning wandering bot: Baltus (81070) class 1 race 5 fac 5, location: mapId 529 WP 5035 'ArathiBasinSpawnH', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3418, minLvl 20, maxLvl 80 (X: 704.07 Y: 704.46 Z: -16.43), flags: 0x00000015 (ArathiBasinSpawnH)
BG bot Baltus id 81070 selected level 53...
Spawning wandering bot: Ken'jai (81071) class 5 race 8 fac 116, location: mapId 529 WP 5035 'ArathiBasinSpawnH', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3418, minLvl 20, maxLvl 80 (X: 704.07 Y: 704.46 Z: -16.43), flags: 0x00000015 (ArathiBasinSpawnH)
BG bot Ken'jai id 81071 selected level 50...
Spawning wandering bot: Rwag (81072) class 4 race 2 fac 2, location: mapId 529 WP 5035 'ArathiBasinSpawnH', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3418, minLvl 20, maxLvl 80 (X: 704.07 Y: 704.46 Z: -16.43), flags: 0x00000015 (ArathiBasinSpawnH)
BG bot Rwag id 81072 selected level 59...
Spawning wandering bot: Tana (81073) class 3 race 10 fac 1610, location: mapId 529 WP 5035 'ArathiBasinSpawnH', 1 link(s) (avg weight 0), Map 529, Zone 3358, Area 3418, minLvl 20, maxLvl 80 (X: 704.07 Y: 704.46 Z: -16.43), flags: 0x00000015 (ArathiBasinSpawnH)
BG bot Tana id 81073 selected level 51...

**Comment by Badgermilk0** [comment]
If I queue SOLO to Warsong Gulch, it works and bots spawn:
```
Spawning wandering bot: Kazi (81074) class 1 race 11 fac 1629, location: mapId 489 WP 4993 'WarsongGulchSpawnA1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4571, minLvl 10, maxLvl 80 (X: 1535.77 Y: 1481.37 Z: 352.11), flags: 0x000000fb (WarsongGulchSpawnA1)
BG bot Kazi id 81074 selected level 60...
Spawning wandering bot: Maxan (81075) class 5 race 3 fac 3, location: mapId 489 WP 4993 'WarsongGulchSpawnA1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4571, minLvl 10, maxLvl 80 (X: 1535.77 Y: 1481.37 Z: 352.11), flags: 0x000000fb (WarsongGulchSpawnA1)
BG bot Maxan id 81075 selected level 57...
Spawning wandering bot: Jol (81076) class 2 race 11 fac 1629, location: mapId 489 WP 4993 'WarsongGulchSpawnA1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4571, minLvl 10, maxLvl 80 (X: 1535.77 Y: 1481.37 Z: 352.11), flags: 0x000000fb (WarsongGulchSpawnA1)
BG bot Jol id 81076 selected level 59...
Spawning wandering bot: Kaerbrus (81077) class 3 race 4 fac 4, location: mapId 489 WP 4993 'WarsongGulchSpawnA1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4571, minLvl 10, maxLvl 80 (X: 1535.77 Y: 1481.37 Z: 352.11), flags: 0x000000fb (WarsongGulchSpawnA1)
BG bot Kaerbrus id 81077 selected level 59...
Spawning wandering bot: Braenna (81078) class 5 race 3 fac 3, location: mapId 489 WP 4993 'WarsongGulchSpawnA1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4571, minLvl 10, maxLvl 80 (X: 1535.77 Y: 1481.37 Z: 352.11), flags: 0x000000fb (WarsongGulchSpawnA1)
BG bot Braenna id 81078 selected level 54...
Spawning wandering bot: Alamar (81079) class 9 race 7 fac 115, location: mapId 489 WP 4993 'WarsongGulchSpawnA1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4571, minLvl 10, maxLvl 80 (X: 1535.77 Y: 1481.37 Z: 352.11), flags: 0x000000fb (WarsongGulchSpawnA1)
BG bot Alamar id 81079 selected level 56...
Spawning wandering bot: Alyissia (81080) class 1 race 4 fac 4, location: mapId 489 WP 4993 'WarsongGulchSpawnA1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4571, minLvl 10, maxLvl 80 (X: 1535.77 Y: 1481.37 Z: 352.11), flags: 0x000000fb (WarsongGulchSpawnA1)
BG bot Alyissia id 81080 selected level 56...
Spawning wandering bot: Thorfin (81081) class 3 race 3 fac 3, location: mapId 489 WP 4993 'WarsongGulchSpawnA1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4571, minLvl 10, maxLvl 80 (X: 1535.77 Y: 1481.37 Z: 352.11), flags: 0x000000fb (WarsongGulchSpawnA1)
BG bot Thorfin id 81081 selected level 52...
Spawning wandering bot: Olmin (81082) class 3 race 3 fac 3, location: mapId 489 WP 4993 'WarsongGulchSpawnA1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4571, minLvl 10, maxLvl 80 (X: 1535.77 Y: 1481.37 Z: 352.11), flags: 0x000000fb (WarsongGulchSpawnA1)
BG bot Olmin id 81082 selected level 57...
Spawning wandering bot: Ayanna (81083) class 3 race 4 fac 4, location: mapId 489 WP 4993 'WarsongGulchSpawnA1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4571, minLvl 10, maxLvl 80 (X: 1535.77 Y: 1481.37 Z: 352.11), flags: 0x000000fb (WarsongGulchSpawnA1)
BG bot Ayanna id 81083 selected level 60...
Spawning wandering bot: Ur'kyo (81084) class 5 race 8 fac 116, location: mapId 489 WP 5002 'WarsongGulchSpawnH1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4572, minLvl 10, maxLvl 80 (X: 919.42 Y: 1434.62 Z: 345.70), flags: 0x000000fd (WarsongGulchSpawnH1)
BG bot Ur'kyo id 81084 selected level 54...
Spawning wandering bot: Cobb (81085) class 5 race 5 fac 5, location: mapId 489 WP 5002 'WarsongGulchSpawnH1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4572, minLvl 10, maxLvl 80 (X: 919.42 Y: 1434.62 Z: 345.70), flags: 0x000000fd (WarsongGulchSpawnH1)
BG bot Cobb id 81085 selected level 57...
Spawning wandering bot: Rupert (81086) class 9 race 5 fac 5, location: mapId 489 WP 5002 'WarsongGulchSpawnH1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4572, minLvl 10, maxLvl 80 (X: 919.42 Y: 1434.62 Z: 345.70), flags: 0x000000fd (WarsongGulchSpawnH1)
BG bot Rupert id 81086 selected level 58...
Spawning wandering bot: Baltus (81087) class 1 race 5 fac 5, location: mapId 489 WP 5002 'WarsongGulchSpawnH1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4572, minLvl 10, maxLvl 80 (X: 919.42 Y: 1434.62 Z: 345.70), flags: 0x000000fd (WarsongGulchSpawnH1)
BG bot Baltus id 81087 selected level 59...
Spawning wandering bot: Mynx (81088) class 6 race 10 fac 1610, location: mapId 489 WP 5002 'WarsongGulchSpawnH1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4572, minLvl 10, maxLvl 80 (X: 919.42 Y: 1434.62 Z: 345.70), flags: 0x000000fd (WarsongGulchSpawnH1)
BG bot Mynx id 81088 selected level 59...
Spawning wandering bot: Meela (81089) class 7 race 6 fac 6, location: mapId 489 WP 5002 'WarsongGulchSpawnH1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4572, minLvl 10, maxLvl 80 (X: 919.42 Y: 1434.62 Z: 345.70), flags: 0x000000fd (WarsongGulchSpawnH1)
BG bot Meela id 81089 selected level 51...
Spawning wandering bot: Gest (81090) class 4 race 2 fac 2, location: mapId 489 WP 5002 'WarsongGulchSpawnH1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4572, minLvl 10, maxLvl 80 (X: 919.42 Y: 1434.62 Z: 345.70), flags: 0x000000fd (WarsongGulchSpawnH1)
BG bot Gest id 81090 selected level 56...
Spawning wandering bot: Quithas (81091) class 8 race 10 fac 1610, location: mapId 489 WP 5002 'WarsongGulchSpawnH1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4572, minLvl 10, maxLvl 80 (X: 919.42 Y: 1434.62 Z: 345.70), flags: 0x000000fd (WarsongGulchSpawnH1)
BG bot Quithas id 81091 selected level 57...
Spawning wandering bot: Urek (81092) class 3 race 6 fac 6, location: mapId 489 WP 5002 'WarsongGulchSpawnH1', 2 link(s) (avg weight 0), Map 489, Zone 3277, Area 4572, minLvl 10, maxLvl 80 (X: 919.42 Y: 1434.62 Z: 345.70), flags: 0x000000fd (WarsongGulchSpawnH1)
BG bot Urek id 81092 selected level 52...
```

**Comment by trickerer** [comment]
Well that clears it. You need to set `NpcBot.Premade.Enable = 0`.
When you queue to a BG in a group of size equal or greater than minimum number of players per team your group is considered premade and core won't match your group against random people, only against another premade group - that's blizzlike behavior. `NpcBot.Premade.Enable`, if enabled, extends this rule to group with NPCbot members as well. If disabled your group will be considered a normal group instead of premade and bots will be generated.
In other words, the behavior you're experiencing is intended.

I thought config description explain that clearly enough but maybe not.

**Comment by Badgermilk0** [comment]
You are correct, the description is good.
Turning it to 0 solves the "issue".

The confusion for this came from the fact that Bot generation DOES happen while "NpcBot.Premade.Enable = 1" for all other BGs except WSG while in a NPCBot Party.

**Comment by trickerer** [comment]
That's because of the minimum number of players per team is different for different BGs. Unfortunately there is no notification or anything when player joins BG queue in premade group.

**Comment by trickerer** [comment]
Another note.
Premade group actually can only wait for match against another premade for a certain amount of time, after which it will be moved to normal groups queue. Default value for this is 30 minutes(!) but it can be changed in config, `Battleground.PremadeGroupWaitForMatch` is the name.

Well then, since the confusion is resolved, closing now.
